src/globus_sdk/transport/requests.py

```diff
@@ -307,6 +307,7 @@ def request(
         headers: dict[str, str] | None = None,
         encoding: str | None = None,
         authorizer: GlobusAuthorizer | None = None,
+        caller_info: RequestCallerInfo | None = None,
```

Comment:
I think we should make this parameter non-nullable -- callers will be required to pass it.

I'd be open to updating the signature to
```
method: str,
url: str,
*,
caller_info: RequestCallerInfo,
query_params: dict[str, t.Any] | None = None,
...
```

so that it's keyword-only and we slightly update the requirement to pass keyword arguments.

Making it required removes a level of conditionality from the checks against it, and I don't think it does us any harm in the base client usage which is responsible for passing this.